### ISSUE TITLE
no-construct-in-interface: oxlint-plugin-awscdk reports `typeof Bucket` (a constructor type) as a Construct type (diverges from eslint-plugin-awscdk)

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#608]
### Describe the bug

`no-construct-in-interface` in oxlint-plugin-awscdk reports a property typed with the **class constructor** (`typeof Bucket`) as if it were a `Bucket` instance. eslint-plugin-awscdk does not report it. The rule exists to prevent exposing a mutable resource *instance* where the read-only interface (`IBucket`) would do; a constructor type has no read-only interface counterpart, so the eslint behaviour is the right one.

`src/props.ts`:

```ts
import { Bucket } from "aws-cdk-lib/aws-s3";

export interface MyConstructProps {
  readonly bucketClass: typeof Bucket;
}
```

Rule enabled: `awscdk/no-construct-in-interface: "error"`.

```
$ npx eslint src/props.ts
(no output)

$ npx oxlint --format=unix -c .oxlintrc.json src/props.ts
src/props.ts:4:3: Interface property 'bucketClass' should not use CDK Construct type 'Bucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]
```

### Expected Behavior

Neither plugin reports `typeof Bucket`.

### Current Behavior

oxlint-plugin-awscdk reports it; eslint-plugin-awscdk does not. The same finder backs `no-construct-in-public-property-of-construct`.

### Plugin

oxlint-plugin-awscdk

### Plugin Version

oxlint-plugin-awscdk 1.0.2 (compared against eslint-plugin-awscdk 4.4.2)

### Language Version

TypeScript 7.0.2 via oxlint-tsgolint 7.0.2001

### Additional Context

```yaml
# machine-readable summary
rules: ["no-construct-in-interface", "no-construct-in-public-property-of-construct"]
symptom: ["false-positive", "cross-plugin-divergence"]
affected_packages: ["oxlint-plugin-awscdk"]
root_cause: "packages/oxlint/src/core/cdk-construct/type-finder/index.ts:13 -> core/ts-type/checker/is-extends-target-super-class.ts:28 (corsa's checker.getBaseTypes() walks the class hierarchy for the constructor type `typeof Bucket` too, so isResourceType / the implements walk succeed on it; in eslint, Type.getBaseTypes() is undefined for a constructor type, which is why eslint stays silent)"
verified_on: "main @ 9caea55 (eslint-plugin-awscdk 4.4.2, oxlint-plugin-awscdk 1.0.2)"
related_issues: ["#572", "#606"]
```

Neither package checks explicitly whether the candidate type is an instance type; eslint is only correct by accident. Verified fix direction, applied to both packages so they stay mirrored: return early from `findTypeOfCdkConstruct` when the type has construct signatures (`type.getConstructSignatures().length` in eslint, `checker.getSignaturesOfType(type, SignatureKind.Construct).length` in oxlint). With that guard the fixture is clean in both plugins, the 22 instance-typed shapes in a wider fixture (`Bucket[]`, `readonly Bucket[]`, tuples, `Record`, `Map`, `Promise`, unions, `Readonly<Bucket>`, ...) are still reported by both, the existing unit tests pass, and `examples/` stays at 43 errors for both linters.

Reproduction environment:

- macOS 14.1, Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.268.0 / constructs 10.8.1
- ESLint 10.9.1 + typescript-eslint 8.69.0 + TypeScript 6.0.3
- oxlint 1.81.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.13.1)
